src/core/model/table/dynamic_table_data.h

```diff
@@ -0,0 +1,82 @@
+#pragma once
+
+#include <cassert>
+#include <string>
+#include <unordered_set>
+#include <vector>
+
+#include "config/exceptions.h"
+#include "config/tabular_data/input_table_type.h"
+
+namespace model {
+
+struct DynamicTableData {
+private:
+    std::vector<std::vector<std::string>> columns_;
+    std::unordered_set<size_t> deleted_rows_{};
+
+public:
+    DynamicTableData(config::InputTable input_table) {
+        columns_.resize(input_table->GetNumberOfColumns());
+        while (input_table->HasNextRow()) {
+            std::vector<std::string> row = input_table->GetNextRow();
+            assert(row.size() == columns_.size());
```

Comment:
`assert` is only used for debugging; this is data from the user, so it should be checked properly.

Reply:
yes, but this case we process during value checking of the `insert` option